### PR TITLE
Update to use new tag method

### PR DESCRIPTION
**Pull request by HarveyLink**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs [hunk 1]
```diff
@@ -657,112 +552,77 @@ private void WriteRemoveTag(CodeWriter writer, ResourceOperation resourceOperati
...
+#pragma warning restore CS8602
+            writer.Append($"{RestClientField}.{CreateMethodName(resourceOperation.GetMethod.Name, async)}( ");
             foreach (string paramNames in pathParamNames)
             {
                 writer.Append($"{paramNames:I}, ");
             }
-            writer.Append($"patchable, cancellationToken)");
+            foreach (Parameter parameter in resourceOperation.GetMethod.RestClientMethod.NonPathParameters)
+            {
+                if (parameter.Name == "expand")
```
Comment (HarveyLink) [inline src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs:609]: Should we hard code here to avoid no expand value error?

File: package-lock.json [hunk 2]
```diff
@@ -1,7 +1,7592 @@
 {
   "name": "@autorest/csharp",
+  "lockfileVersion": 2,
```
Comment (m-nash) [inline package-lock.json:3]: are these changes intended?

File: package.json [hunk 3]
```diff
@@ -4,6 +4,6 @@
   "description": "package.json intended for in-repo use only, package.json used for publishing is located in src/AutoRest.CSharp/package.json",
   "devDependencies": {
     "@microsoft.azure/autorest.testserver": "3.0.21",
-    "autorest": "3.2.1"
+    "autorest": "3.2.3"
```
Comment (m-nash) [inline package.json:7]: is it necessary to update autorest version for this?

File: samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionVirtualMachineScaleSet.cs [hunk 4]
```diff
@@ -11,13 +11,13 @@
 
 namespace Azure.ResourceManager.Sample
 {
-    /// <summary> A Class representing a VirtualMachineExtensionVirtualMachineScaleSets along with the instance operations that can be performed on it. </summary>
-    public class VirtualMachineExtensionVirtualMachineScaleSets : VirtualMachineExtensionVirtualMachineScaleSetsOperations
+    /// <summary> A Class representing a VirtualMachineExtensionVirtualMachineScaleSet along with the instance operations that can be performed on it. </summary>
```
Comment (m-nash) [inline samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionVirtualMachineScaleSet.cs:14]: looks like you might need to sync the latest feature/v3

**Review by m-nash — CHANGES_REQUESTED** [review]

File: samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionVirtualMachineScaleSet.cs (hunk 4 above)
Comment (HarveyLink) [reply samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionVirtualMachineScaleSet.cs]: Done

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs [hunk 5]
```diff
@@ -657,112 +552,77 @@ private void WriteRemoveTag(CodeWriter writer, ResourceOperation resourceOperati
...
+#pragma warning restore CS8602
+            writer.Append($"{RestClientField}.{CreateMethodName(resourceOperation.GetMethod.Name, async)}( ");
             foreach (string paramNames in pathParamNames)
             {
                 writer.Append($"{paramNames:I}, ");
             }
-            writer.Append($"patchable, cancellationToken)");
+            foreach (Parameter parameter in resourceOperation.GetMethod.RestClientMethod.NonPathParameters)
+            {
+                if (parameter.ValidateNotNull)
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs:609]: Remove change

File: package-lock.json (hunk 2 above)
Comment (HarveyLink) [reply package-lock.json]: Remove change

File: package.json (hunk 3 above)
Comment (HarveyLink) [reply package.json]: Remove change

**Review by m-nash — APPROVED** [review]
